implementations/rust/ockam/ockam/src/node.rs

```diff
@@ -0,0 +1,329 @@
+use crate::remote::{RemoteForwarder, RemoteForwarderInfo, RemoteForwarderOptions};
+use crate::stream::Stream;
+use core::time::Duration;
+use ockam_core::compat::string::String;
+use ockam_core::compat::sync::Arc;
+use ockam_core::{
+    Address, AsyncTryClone, IncomingAccessControl, Message, OutgoingAccessControl, Processor,
+    Result, Route, Routed, Worker,
+};
+use ockam_identity::{
+    secure_channels, Credentials, CredentialsServer, Identities, IdentitiesCreation,
+    IdentitiesKeys, IdentitiesRepository, SecureChannelRegistry, SecureChannels,
+    SecureChannelsBuilder, Storage,
+};
+use ockam_identity::{
+    IdentitiesVault, Identity, SecureChannelListenerOptions, SecureChannelOptions,
+};
+use ockam_node::{Context, MessageReceiveOptions, MessageSendReceiveOptions};
+#[cfg(feature = "std")]
+use ockam_transport_tcp::TcpTransport;
+
+/// This struct supports all the ockam services for managing identities
+/// and creating secure channels
+pub struct Node {
+    context: Context,
+    secure_channels: Arc<SecureChannels>,
+}
+
+/// Create a default node (with no persistence)
+pub fn node(ctx: Context) -> Node {
+    Node {
+        context: ctx,
+        secure_channels: secure_channels(),
+    }
+}
+
+/// Return a new builder for top-level services
+pub fn builder() -> NodeBuilder {
+    NodeBuilder::new()
+}
```

Comment:
`node` is exposed like that because for now it is used directly in examples as 
```rust
#[ockam_macros::node]
fn main(ctx: Context) -> Result<()> {
  let node = node(&ctx);
...
}
```

With the idea that later on we could simply have
```
#[ockam_macros::node]
fn main(node: Node) -> Result<()> {
...
}
```

> Also, we have been attaching the builder methods to the built struct in several places of the codebase.

Ok, I can do this.

Reply:
Done (the `builder` move)

Reply:
BTW the `node` function is mostly meant to be used in examples to keep them short. If you look at the changed code in `ockam_api` it is using `builder`.